src/crimson/osd/ops_executer.cc

```diff
@@ -1089,11 +1089,15 @@ ObjectContextRef OpsExecuter::prepare_clone(
   clone_obs.oi.copy_user_bits(initial_obs.oi);
   clone_obs.oi.clear_flag(object_info_t::FLAG_WHITEOUT);
 
-  auto [clone_obc, existed] = pg->obc_registry.get_cached_obc(std::move(coid));
-  ceph_assert(!existed);
-
+  auto [clone_obc, existed] = pg->obc_registry.get_cached_obc(coid);
+  (void)existed; // cache hits are valid; do not assert on them
```

Comment:
Nit: `[[maybe_unused]] auto [clone_obc, existed]` might be slightly more conventional.